### PR TITLE
feat: Verify blobs support in MEV-Boost/Relay/Builder forks

### LINKED ISSUE #113: Support blobs in MEV-Boost/Relay/Builder forks
There is already an issue to support blobs in the Bolt sidecar: #105. However, some work is also needed in the rest of the pipeline to make sure the builder is able to create bid with pre-confirmed blob transaction.

Ideally, this should be done by not making any breaking change to the specs.

```[tasklist]
### Tasks
- [ ] Support blobs in MEV-Boost fork
- [ ] Support blobs in Flashbots' relay fork
- [ ] Support blobs in Flashbots' builder fork
```

### PR DESCRIPTION
**Pull request by namn-grg**
fixes #113 

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved error logging for Merkle multiproofs calculation to provide more detailed messages.
  - Refinement of transaction handling and error messages during Merkle proof calculations.

- **New Features**
  - Added support for creating and processing sample blob transactions in block building tests.
  - Enhanced constraints handling for specific slots in tests.

- **Tests**
  - Updated test logic for block building and transaction data handling.
  - Added new import and transaction data modifications in test scenarios.

- **Refactor**
  - Adjusted signature assignment in test functions for better compatibility with `BLSSignature` method.

### CODE REVIEW

File: builder/builder/utils.go [hunk 1]
```diff
@@ -161,7 +161,7 @@ func CalculateMerkleMultiProofs(
 	// BOLT: generate merkle tree from payload transactions (we need raw RLP bytes for this)
 	rawTxs := make([]bellatrix.Transaction, len(payloadTransactions))
 	for i, tx := range payloadTransactions {
-		raw, err := tx.MarshalBinary()
+		raw, err := tx.WithoutBlobTxSidecar().MarshalBinary()
```
Comment (merklefruit) [inline builder/builder/utils.go:164]: didn't know of this method. awesome!

File: mev-boost/server/service_test.go [hunk 2]
```diff
@@ -348,8 +348,8 @@ func TestConstraintsAndProofs(t *testing.T) {
 			Slot:           slot,
 			Constraints:    []*Constraint{{Transaction(rawTx), nil}},
 		},
-		Signature: _HexToBytes(
-			"0x81510b571e22f89d1697545aac01c9ad0c1e7a3e778b3078bef524efae14990e58a6e960a152abd49de2e18d7fd3081c15d5c25867ccfad3d47beef6b39ac24b6b9fbf2cfa91c88f67aff750438a6841ec9e4a06a94ae41410c4f97b75ab284c"),
+		Signature: phase0.BLSSignature(_HexToBytes(
+			"0x81510b571e22f89d1697545aac01c9ad0c1e7a3e778b3078bef524efae14990e58a6e960a152abd49de2e18d7fd3081c15d5c25867ccfad3d47beef6b39ac24b6b9fbf2cfa91c88f67aff750438a6841ec9e4a06a94ae41410c4f97b75ab284c")),
```
Comment (namn-grg) [inline mev-boost/server/service_test.go:352]: Coincidentally saw this error and fixed it.

**Review by merklefruit — APPROVED** [review]
Really great job with this test. this is super useful!

**Review by thedevbirb — APPROVED** [review]
Awesome! Love that it required so few changes

**Pull request merged**